gprofiler/profilers/dotnet.py

```diff
@@ -0,0 +1,157 @@
+        version = self.get_exe_version(process)  # not using cached version here since this wrapper is a cache
+        return version
+
+    def make_application_metadata(self, process: Process) -> Dict[str, Any]:
+        # ruby version
+        version = self._get_dotnet_version(process)
+
+        metadata = {"dotnet_version": version}
+
+        metadata.update(super().make_application_metadata(process))
+        return metadata
+
+
+@register_profiler(
+    "dotnet",
+    possible_modes=["dotnet-trace", "disabled"],
+    supported_archs=["x86_64", "aarch64"],
+    default_mode="dotnet-trace",
+)
+class DotnetProfiler(ProcessProfilerBase):
+
+    RESOURCE_PATH = "dotnet/tools/dotnet-trace"
+    MAX_FREQUENCY = 100
+    _EXTRA_TIMEOUT = 10
+
+    def __init__(
+        self,
+        frequency: int,
+        duration: int,
+        stop_event: Optional[Event],
+        storage_dir: str,
+        profile_spawned_processes: bool,
+        dotnet_mode: str,
+    ):
+        super().__init__(frequency, duration, stop_event, storage_dir)
+        assert dotnet_mode == "dotnet-trace", \
+                              "Dotnet profiler should not be initialized, wrong dotnet-trace value given"
+        self._metadata = DotnetMetadata(self._stop_event)
+
+    def _make_command(self, process: Process, duration: int) -> List[str]:
+        result = run_process(
+            f"ls /proc/{process.pid}/root/tmp | grep diagnostic ",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True,
+            suppress_log=True,
+            check=False,
+        )
+        socket_list = result.stdout.decode("utf-8").split('\n')
+        tempdir = f"TMPDIR=/proc/{process.pid}/root/tmp"
+        internal_pid = str(process.pid)
+        metadata = None
+        if int(socket_list[0].split("-")[2]) != 1:
+            # we are not in docker, pid different than 1
+            metadata = self._metadata.get_metadata(process)
+        else:
+            # we are in docker, pid is 1
+            internal_pid = "1"
+        return [
+            "sudo",
+            tempdir,
+            resource_path(self.RESOURCE_PATH),
+            "collect",
+            "--format",
+            "speedscope",
+            "--process-id",
+            internal_pid,
+            "--duration",
+            self._parse_duration(duration),
+            #           str(self._frequency), - TODO: frequency handling to be determined
+        ], metadata
+
+    def _profile_process(self, process: Process, duration: int, spawned: bool) -> ProfileData:
+        logger.info(
+            f"Profiling{' spawned' if spawned else ''} process {process.pid} with dotnet-trace",
+            cmdline=" ".join(process.cmdline()),
+            no_extra_to_server=True,
+        )
+        appid = None
+        app_metadata = None
+        local_output_path = os.path.join(self._storage_dir, f"dotnet-trace.{random_prefix()}.{process.pid}.col")
```

Comment:
Why assign it if you override later? Is there no way to ask dotnet-trace to write to this path?